Library/DasharoSystemFeaturesUiLib/DasharoSystemFeatures.c

```diff
@@ -230,10 +226,28 @@ DasharoSystemFeaturesUiLibConstructor (
         &mMeModeDefault
         );
     mDasharoSystemFeaturesPrivate.DasharoFeaturesData.MeMode = mMeModeDefault;
+    ASSERT_EFI_ERROR (Status);
   }
 
-  if (EFI_ERROR(Status)) {
-    return Status;
+  BufferSize = sizeof (mDasharoSystemFeaturesPrivate.DasharoFeaturesData.Ps2Controller);

```

Comment:
This function is becoming a little crowded :\ maybe this isn't the place for this but at some point it should probably be cleaned up

Reply:
I was thinking about it as well. Certain options have a pattern and it could be handled with a single define. I can upload a PR later to squeeze it a little bit.